### PR TITLE
[TEST] Add codeclimate config yml

### PR DESCRIPTION
**Pull request by eoghanmcilwaine**
LRN-22775

## Checklist

- [ ] Feature
- [ ] Bug
- [ ] Security
- [ ] Documentation

- [ ] ChangeLog.md updated

- [ ] Tests added
- [ ] All testsuites passed
- [x] `make dist` completed successfully

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+version: "2"
+plugins:
+  radon:
+    enabled: true
+    config:
+      threshold: "C"
+  pep8:
```
Comment (shtrom) [inline .codeclimate.yml]: We might want to import a linter such as flake8 or autopep, and runt that as part of a `test-lint` target that would be a dependency of `test-unit`

File: .codeclimate.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    enabled: true
+  duplication:
+    enabled: true
+    config:
+      languages:
+      - python:
+exclude_patterns:
+- tests/**/*
```
Comment (shtrom) [inline .codeclimate.yml]: Well, we should have tests of as good a quality as the rest of our code. Or is it already a lost cause?

Comment (eoghanmcilwaine) [reply .codeclimate.yml]: Fair enough, will remove this.

File: .codeclimate.yml (hunk 1 above)
Comment (eoghanmcilwaine) [reply .codeclimate.yml]: Sounds like a good idea for a future improvement. These ones here are just official code climate analysis plugins though. https://docs.codeclimate.com/docs/list-of-engines

Comment (shtrom) [reply .codeclimate.yml]: Oh, not for CodeClimate. I meant adding those check to our local toolchain so we can match the PEP8 requirement more easily, without needing a round-trip to CodeClimate.

Comment (eoghanmcilwaine) [reply .codeclimate.yml]: Sure. It's out of scope for this task though.

**Pull request merged**